packages/eslint-plugin/src/rules/no-array-delete.ts

```diff
@@ -0,0 +1,92 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import { unionTypeParts } from 'tsutils';
+import * as ts from 'typescript';
+
+import * as util from '../util';
+
+export default util.createRule({
+  name: 'no-array-delete',
+  meta: {
+    hasSuggestions: true,
+    docs: {
+      description: 'Disallow delete operator for arrays',
+      recommended: 'strict',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      arrayDelete: 'Using the delete operator on an array is dangerous.',
+    },
+    schema: [],
+    type: 'problem',
+    fixable: 'code',
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      "UnaryExpression[operator='delete'] > MemberExpression[computed]"(
+        node: TSESTree.MemberExpressionComputedName & {
+          parent: TSESTree.UnaryExpression;
+        },
+      ): void {
+        const parserServices = util.getParserServices(context);
+        const checker = parserServices.program.getTypeChecker();
+        const originalNode = parserServices.esTreeNodeToTSNodeMap.get(node);
+
+        const target = originalNode.getChildAt(0);
+        const key = originalNode.getChildAt(2);
+
+        const targetType = util.getConstrainedTypeAtLocation(checker, target);
+
+        if (!util.isTypeAnyArrayType) {
```

Comment:
are you sure about this?

Reply:
better solution? I'm not an expert, if you notice some better options i would like to know...

Reply:
`isTypeAnyArrayType` is a function, but you are not calling it you are just checking if its present

Reply:
HOLY CRAP, thank you. I DID NOT WRITE CODE LIKE THIS IN ANY PERIOD OF MY LIFE.